src/evm_utils/proof_serde.rs

```diff
@@ -111,12 +89,47 @@ impl Serialize for OpenZeppelinMultiProof {
                 .collect::<Vec<String>>(),
         )?;
         state.serialize_field("proofFlags", &self.proof_flags)?;
+
+        state.end()
+    }
+}
+
+impl<F: PrimeField> Serialize for WhirEvmProof<F> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("WhirEvmProof", 1)?;
         state.serialize_field(
-            "root",
-            // "0x" is necessary for Foundry to recognize it as bytes32
-            &("0x".to_owned() + &hex::encode(self.root.as_ref())),
+            "proofs",
+            &self
+                .0
+                .iter()
+                .map(|(proof, answers)| {
+                    (
+                        proof,
+                        answers
+                            .iter()
+                            .map(|inner_vec| inner_vec.iter().map(F::serialize).collect())
+                            .collect::<Vec<Vec<String>>>(),
+                    )
+                })
+                .collect::<Vec<_>>(),
         )?;
-
         state.end()
     }
 }
+
+pub trait EvmFieldElementSerDe {
+    /// Serialize a PrimeField to a hex string in EVM format
+    fn serialize(&self) -> String;
+}
+
+impl<T: PrimeField> EvmFieldElementSerDe for T {
```

Comment:
Hex format of uint256 is more versatile for foundry - you can do both individual value deserialization and whole struct deserialization, while string-style (e.g., "123") only works with individual value deserialization.